src/building/menu.c

```diff
@@ -1,13 +1,12 @@
 #include "menu.h"
 
 #include "city/buildings.h"
+#include "core/config.h"
 #include "empire/city.h"
 #include "game/tutorial.h"
 #include "scenario/building.h"
 
-#define MAX_BUILDING_ITEMS 30
-
-static const building_type MENU_BUILDING_TYPE[MAX_BUILDING_ITEMS][MAX_BUILDING_ITEMS] = {
+static const building_type MENU_BUILDING_TYPE[BUILD_MENU_MAX][BUILD_MENU_MAX] = {
```

Comment:
The second dimension should not be BUILD_MENU_MAX, since it contains buildings rather than build menus. If we remove the unused menu items later on, this may cause an out-of-bounds error. Same goes for the enabled variable, and the loop variables.